app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -48,28 +48,51 @@ def inherited_attributes
           inherited_attrs
         end
 
-        def apply_inherited_attributes(attributes, initialized = true)
-          attributes = super(attributes, initialized) || {}
-          facet_attrs = attributes&.[]('content_facet_attributes')
-          return attributes if facet_attrs.blank?
-          cv_id = facet_attrs['content_view_id']
-          lce_id = facet_attrs['lifecycle_environment_id']
-          if initialized && (cv_id.blank? || lce_id.blank?)
-            if cv_id.blank?
-              Rails.logger.info "Hostgroup has no content view assigned; using host's existing content view"
-              facet_attrs['content_view_id'] = content_facet&.single_content_view&.id
-            end
-            if lce_id.blank?
-              Rails.logger.info "Hostgroup has no lifecycle environment assigned; using host's existing lifecycle environment"
-              facet_attrs['lifecycle_environment_id'] = content_facet&.single_lifecycle_environment&.id
-            end
-            attributes['content_facet_attributes'] = facet_attrs
-          else
-            Rails.logger.debug "Hostgroup has content view and lifecycle environment assigned; using those"
+        def apply_inherited_attributes(input_attributes, initialized = true)
+          attributes = super(input_attributes.clone, initialized) || {}
+
+          # When the input explicitly sets content_facet_attributes, keep them
+          # regardless of what any higher function calls added
+          if initialized && input_attributes.key?(:content_facet_attributes) &&
+            attributes.key?(:content_facet_attributes)
```

Comment:
This line formatting is confusing. I'd expect RuboCop to disagree with it, because it gives the impression to be part of the code block rather than the condition.

Reply:
Thank you for the coding style notes, I appreciate it. I'm going to resolve these since the source has changed so much in the last few days due to new information.